cognite/client/_api_client.py

```diff
@@ -230,21 +213,27 @@ def _get_base_url_with_base_path(self) -> str:
 
     def _is_retryable(self, method: str, path: str) -> bool:
         valid_methods = ["GET", "POST", "PUT", "DELETE", "PATCH"]
-        match = re.match(
-            r"(?:http|https)://[a-z\d.:\-]+(?:/api/(?:v1|playground)/projects/[^/]+)?(/[^\?]+)?(\?.+)?", path
-        )
-        if not match:
-            raise ValueError("Path {} is not valid. Cannot resolve whether or not it is retryable".format(path))
+
         if method not in valid_methods:
             raise ValueError("Method {} is not valid. Must be one of {}".format(method, valid_methods))
-        path_end = match.group(1)
 
         if method in ["GET", "PUT", "PATCH"]:
             return True
-        if method == "POST" and path_end in self.RETRYABLE_POST_ENDPOINTS:
+        if method == "POST" and self._endpoint_path_is_retryable(path):
             return True
         return False
 
+    @classmethod
+    @functools.lru_cache(64)
+    def _endpoint_path_is_retryable(cls, path: str) -> bool:
+        valid_path_pattern = r"(?:http|https)://[a-z\d.:\-]+(?:/api/(?:v1|playground)/projects/[^/]+)?(/[^\?]+)?(\?.+)?"
+        if not re.match(valid_path_pattern, path):
+            raise ValueError("Path {} is not valid. Cannot resolve whether or not it is retryable".format(path))
+        for pattern in cls._RETRYABLE_POST_ENDPOINT_REGEX_PATTERNS:
+            if re.search(pattern, path):
```

Comment:
```suggestion
            if re.search("^" + pattern + "$", path):
```
or something like that to exclude extra stuff?

Reply:
That wont work since urls can include query params after and different protocols/hosts/ports before.

Reply:
but here it's path, so prefix part should be ok. Suffix yes, probably then something like `[#?].*$
just the prefix part should be ok more or less as working urls aren't prefixes of each other (mostly)